### PR TITLE
CronJob support

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

File: controllers/cloud.redhat.com/makers/maker.go [hunk 1]
```diff
@@ -191,6 +200,201 @@ func applyPodAntiAffinity(t *core.PodTemplateSpec) {
...
+	initJob(m.App, m.Env, &c, nn, job, hash)
+
+	if err := update.Apply(m.Ctx, m.Client, &c); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func initJob(app *crd.ClowdApp, env *crd.ClowdEnvironment, jc *batch.CronJob, nn types.NamespacedName, job crd.Job, hash string) {
```
Comment (kylape) [inline controllers/cloud.redhat.com/makers/maker.go:227]: Should `jc` be `cj`?

File: controllers/cloud.redhat.com/makers/maker.go [hunk 2]
```diff
@@ -191,6 +200,201 @@ func applyPodAntiAffinity(t *core.PodTemplateSpec) {
...
+	labels := app.GetLabels()
+	labels["pod"] = nn.Name
+	app.SetObjectMeta(jc, crd.Name(nn.Name), crd.Labels(labels))
+
+	pod := job.PodSpec
+
+	jc.Spec.Schedule = job.Schedule
```
Comment (kylape) [inline controllers/cloud.redhat.com/makers/maker.go:234]: If we are assuming that the `Schedule` will always be set on the spec, we should not configure the field with `omitempty`.  We can always add this back later when we support the more generic `Job` resource.

File: controllers/cloud.redhat.com/makers/maker.go [hunk 3]
```diff
@@ -191,6 +200,201 @@ func applyPodAntiAffinity(t *core.PodTemplateSpec) {
...
+
+	jc.Spec.JobTemplate.ObjectMeta.Labels = labels
+	jc.Spec.JobTemplate.Spec.Template.ObjectMeta.Labels = labels
+	jc.Spec.JobTemplate.Spec.Template.Spec.RestartPolicy = core.RestartPolicyNever
```
Comment (kylape) [inline controllers/cloud.redhat.com/makers/maker.go:238]: Should we make this configurable?

File: controllers/cloud.redhat.com/makers/maker.go [hunk 4]
```diff
@@ -191,6 +200,201 @@ func applyPodAntiAffinity(t *core.PodTemplateSpec) {
...
+		readinessProbe = *pod.ReadinessProbe
+	} else {
+		readinessProbe = core.Probe{}
+	}
+
+	var lcpu, lmemory, rcpu, rmemory resource.Quantity
+	nullCPU := resource.Quantity{Format: resource.DecimalSI}
+	nullMemory := resource.Quantity{Format: resource.BinarySI}
+
+	if *pod.Resources.Limits.Cpu() != nullCPU {
```
Comment (kylape) [inline controllers/cloud.redhat.com/makers/maker.go:265]: Can this block of code be reused between this and `initDeployment`?

File: controllers/cloud.redhat.com/makers/maker.go [hunk 5]
```diff
@@ -191,6 +200,201 @@ func applyPodAntiAffinity(t *core.PodTemplateSpec) {
...
+		MountPath: "/cdapp/",
+	})
+
+	jc.Spec.JobTemplate.Spec.Template.Spec.Containers = []core.Container{c}
+
+	if len(pod.InitContainers) > 0 {
+		jc.Spec.JobTemplate.Spec.Template.Spec.InitContainers = make([]core.Container, len(pod.InitContainers))
+	}
+
+	for i, ic := range pod.InitContainers {
```
Comment (kylape) [inline controllers/cloud.redhat.com/makers/maker.go:333]: Similar comment here.  Could this be put into a common function to be used both here and `initDeployment`?

**Review by kylape — CHANGES_REQUESTED** [review]

File: controllers/cloud.redhat.com/makers/maker.go (hunk 2 above)
Comment (psav) [reply controllers/cloud.redhat.com/makers/maker.go:234]: Yeh, that was why I made it optional to allow the use of specifying Jobs

Comment (psav) [reply controllers/cloud.redhat.com/makers/maker.go:234]: But you are right

File: controllers/cloud.redhat.com/makers/maker.go (hunk 3 above)
Comment (psav) [reply controllers/cloud.redhat.com/makers/maker.go:238]: Possibly so - yes - I didn't think always was appropriate - and I wondered about on failure too

File: controllers/cloud.redhat.com/makers/maker.go (hunk 4 above)
Comment (psav) [reply controllers/cloud.redhat.com/makers/maker.go:265]: I'll DRY it up

File: controllers/cloud.redhat.com/makers/maker.go (hunk 5 above)
Comment (psav) [reply controllers/cloud.redhat.com/makers/maker.go:333]: I'll DRY it up

**Review by kylape — APPROVED** [review]

**Pull request merged**